### PR TITLE
Uses CustomEvents within ImportJobs

### PR DESCRIPTION
**Pull request by andyHa**
### Description

Makes import jobs dispatch custom events.

grafik

grafik

grafik

Example Scripts:
```
registry.registerHandler(sirius.biz.jobs.batch.ImportJobStartedEvent.class,  |e| log(e));

registry.registerTypedHandler(sirius.biz.importer.BeforeFindEvent.class, MongoUserAccount.class, |e| log(e));

registry.registerTypedHandler(sirius.biz.importer.OnLoadEvent.class, MongoUserAccount.class, |e| log(e));

registry.registerTypedHandler(sirius.biz.importer.BeforeCreateOrUpdateEntityEvent.class, MongoUserAccount.class, |e| log(e));
```

Relates to:
- https://github.com/scireum/sirius-biz/pull/1959
- https://github.com/scireum/sirius-biz/pull/1960
- https://github.com/scireum/sirius-biz/pull/1961

### Checklist

- [x] Code change has been tested and works locally
- [x] Code was formatted via IntelliJ and follows SonarLint & [best practices](https://scireum.myjetbrains.com/youtrack/articles/MISC-A-16/CodeStyle-JavaDoc)
- [x] Patch Tasks: Is local execution of Patch Tasks necessary? If so, please also mark the PR with the tag.

### CODE REVIEW

**Review by idlira — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/biz/importer/OnLoadEvent.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Triggered within {@link sirius.biz.importer.ImportHandler#load(Context, BaseEntity)} in order to update an entity
+ * using the given context.
+ *
+ * @param <E> the type of entity being updated
+ */
+public class OnLoadEvent<E extends Entity> extends ContextScriptableEvent<E> {
```
Comment (idlira) [inline src/main/java/sirius/biz/importer/OnLoadEvent.java:25]: I would name it BeforeLoadEvent, just in case we ever need an AfterLoadEvent later on...

Comment (sabieber) [reply src/main/java/sirius/biz/importer/OnLoadEvent.java:25]: But its currently invoked after the mappings are loaded

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/biz/importer/OnLoadEvent.java (hunk 1 above)
Comment (idlira) [reply src/main/java/sirius/biz/importer/OnLoadEvent.java:25]: Yes. Just the opposite of what I said: rename to AfterLoadEvent so we can have a BeforeLoadEvent.
Besides, as to the current known use-cases we need to change the context BEFORE the load in most of the cases.

Comment (andyHa) [reply src/main/java/sirius/biz/importer/OnLoadEvent.java]: Naming has been unified and split into BeforeLoad and AfterLoad to support all use-cases (both events receive the entity and the context, so the data can be changed on either side (depending on the evnt type))

**Review by fhaScireum — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by idlira — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Pull request merged**